packages/transaction-controller/src/TransactionController.ts

```diff
@@ -610,7 +610,10 @@ export class TransactionController extends BaseControllerV1<
       getEthQuery: () => this.ethQuery,
       getTransactions: () => this.state.transactions,
       isResubmitEnabled: pendingTransactions.isResubmitEnabled,
-      nonceTracker: this.nonceTracker,
+      getGlobalLock: async () =>
+        this.#acquireNonceLockForChainIdKey({
+          chainId: this.getChainId(),
+        }),
```

Comment:
As written what happens when the **first** `onLatestBlock` event happens for two clients with the same chainId? Do they both race to add the mutex and then acquire it? I guess there isn't any issue with them overwriting each other in `nonceMutexesByChainId` since the keys will be the same? And then one of them wins and the other waits... Yeah I guess this works as is.